File: src/backend/WordsmithHub.API/Features/Freelances/Models/FreelanceDto.cs
using WordsmithHub.API.Features.Common.Models;

namespace WordsmithHub.API.Features.Freelances.Models;

public record FreelanceDto
{
    public required Guid Id { get; set; }
    public required string FirstName { get; set; }
    public required string LastName { get; set; }
    public string? Phone { get; set; }
    public required string Email { get; set; }
    public required AddressDto? Address { get; set; }
    public required int StatusId { get; set; }
}

File: src/gui/src/profile/models/FreelanceDto.ts
import type { Address } from "../../models/Address";

export default class FreelanceDto {
    id: string;
    firstName: string;
    lastName: string;
    email: string;
    phone: string | null;
    address: Address | null;
    statusId?: number;

    public constructor(
        id: string,
        firstName: string,
        lastName: string,
        email: string,
        phone: string | null = null,
        address: Address | null = null,
        statusId?: number
    ) {
        this.id = id;
        this.firstName = firstName;
        this.lastName = lastName;
        this.email = email;
        this.phone = phone;
        this.address = address;
        this.statusId = statusId;
    }
}